src/bootstrap/builder.rs

```diff
@@ -759,7 +765,8 @@ impl<'a> Builder<'a> {
         };
 
         let mut rustflags = Rustflags::new(&target);
-        if stage != 0 {
+        // nightly clippy needs --cfg=bootstrap to run successfully
+        if stage != 0 || cmd == "clippy" {
```

Comment:
I am actually rather surprised by us needing to do this, as AFAICT if the cmd is clippy we should always be in stage 0? When is that not the case today?

(and if Clippy is already in stage 0, then the right flags would be in RUSTFLAGS_BOOTSTRAP, right?)

Reply:
You are right, my code comment is simply wrong.
Looking at `--verbose` output, we are indeed in `stage 0` all the time.
Looking at the error I was getting previously
```
error[E0061]: this function takes 1 argument but 0 arguments were supplied
  --> src/librustc_data_structures/box_region.rs:87:52
   |
87 |         let result = Pin::new(&mut self.generator).resume();
   |                                                    ^^^^^^- supplied 0 arguments
   |
help: expected the unit value `()`; create it with empty parentheses
   |
87 |         let result = Pin::new(&mut self.generator).resume(());
   |                                                           ^^
````

and the corresponding code 
````rust

    #[cfg(bootstrap)]
    pub fn complete(&mut self) -> R {
        // Tell the generator we want it to complete, consuming it and yielding a result
        BOX_REGION_ARG.with(|i| i.set(Action::Complete));

        let result = Pin::new(&mut self.generator).resume();
        if let GeneratorState::Complete(r) = result { r } else { panic!() }
    }

    #[cfg(not(bootstrap))]
    pub fn complete(&mut self) -> R {
        // Tell the generator we want it to complete, consuming it and yielding a result
        BOX_REGION_ARG.with(|i| i.set(Action::Complete));

        let result = Pin::new(&mut self.generator).resume(());
        if let GeneratorState::Complete(r) = result { r } else { panic!() }
    }
````
we actually need to disable cfg=booststrap for `x.py clippy` in order to have it work from what it seems.